From: Duan Jiong <djduanjiong@gmail.com>
Date: Sat, 8 Sep 2012 16:32:28 +0000
Subject: [PATCH] etherdevice: introduce help function eth_zero_addr()
Origin: https://git.kernel.org/linus/6d57e9078e880a3dd232d579f42ac437a8f1ef7b

a lot of code has either the memset or an inefficient copy
from a static array that contains the all-zeros Ethernet address.
Introduce help function eth_zero_addr() to fill an address with
all zeros, making the code clearer and allowing us to get rid of
some constant arrays.

Signed-off-by: Duan Jiong <djduanjiong@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 include/linux/etherdevice.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/linux/etherdevice.h b/include/linux/etherdevice.h
index 0ca91de..71929cd 100644
--- a/include/linux/etherdevice.h
+++ b/include/linux/etherdevice.h
@@ -154,6 +154,17 @@ static inline void dev_hw_addr_random(struct net_device *dev, u8 *hwaddr)
 }
 
 /**
+ * eth_zero_addr - Assign zero address
+ * @addr: Pointer to a six-byte array containing the Ethernet address
+ *
+ * Assign the zero address to the given address array.
+ */
+static inline void eth_zero_addr(u8 *addr)
+{
+	memset(addr, 0x00, ETH_ALEN);
+}
+
+/**
  * eth_hw_addr_random - Generate software assigned random Ethernet and
  * set device flag
  * @dev: pointer to net_device structure
